keps/sig-scheduling/3280-guarantee-pdb-when-preemption-happens/README.md

```diff
@@ -0,0 +1,973 @@
+during the scheduler preemption process, in order to meet the needs of users who do not want PDB violations to occur 
+during scheduler preemption due to some reasons such as high availability of 
+services(https://github.com/kubernetes/kubernetes/issues/91492#issuecomment-1029484252). 
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+`PodDisruptionBudget` (PDB) is used to limit the number of concurrent disruptions that your application experiences, 
+allowing for high availability. Users can set the field `.spec.maxUnavailable` or `.spec.minAvailable` to declare 
+the current minimum availability or maximum unavailability to be maintained after eviction. 
+
+However, there is currently an issue where the kube-scheduler does not strictly guarantee PDBs during the preemption 
+process. The scheduler supports PDBs when preempting pods, but the adherence to PDBs is best effort. The scheduler 
+attempts to select victims whose PDBs are not violated during preemption, but if no such victims are found, preemption 
+will still take place, resulting in the removal of lower-priority pods despite their PDBs being violated.
+
+PodDisruptionBudgets (PDBs) are frequently used for stability and the possibility of violating a PDB during the preemption 
+process is not acceptable for certain users. As such, it is beneficial to provide users with the option to choose if they 
+want the PDB to be guaranteed during preemption or not.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Provide an optional option to configure whether the scheduler needs to make `PodDisruptionBudget` 
+  guaranteed when preemption happens.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+I deployed a service in a cluster and to ensure its high availability, 
```

Comment:
Stories are usually written in 3rd person, ie. User deploys a service...

Reply:
Done